### PR TITLE
feat: add component `CCardDialog`

### PR DESCRIPTION
**Pull request by ItsPierre**
Requires #47 and #52

### CODE REVIEW

File: src/components/CCard.vue [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+);
+</script>
+
+<style scoped lang="scss">
+.c-card .c-card-sections > * {
+  &:not(:first-child):not(:last-child),
+  &:first-child:not(:last-child) {
+    margin-bottom: 12px;
+  }
+}
```
Comment (johannes-huther) [inline src/components/CCard.vue]: If I undestand that correctly, you want to apply the margin to every child of `.c-card-sections`, if

- it is not the first and not the last child
- if it is the first and not the last child

If so, wouldn't that be enough?

```suggestion
.c-card .c-card-sections > * {
  &:not(:last-child) {
    margin-bottom: 12px;
  }
}
```

**Review by johannes-huther — CHANGES_REQUESTED** [review]

File: src/components/CCard.vue (hunk 1 above)
Comment (ItsPierre) [reply src/components/CCard.vue]: this is from another commit, this will be fixed as soon as #47 is merged and this PR is rebases

Comment (johannes-huther) [reply src/components/CCard.vue]: oh I got confused with the PRs :D

File: src/components/CCardDialog.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    loading?: boolean;
+  }>(),
+  {
+    title: undefined,
+    subtitle: undefined,
+    width: "500px",
+    maxWidth: undefined,
+    minWidth: undefined,
+    persistent: false,
```
Comment (johannes-huther) [inline src/components/CCardDialog.vue:88]: are the `undefined`s actually necessary?

File: src/components/CCardDialog.vue [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  },
+);
+</script>
+
+<style lang="scss">
+.c-dialog {
+  .v-overlay__content {
+    margin: 0 !important;
+    max-width: 100% !important;
```
Comment (johannes-huther) [inline src/components/CCardDialog.vue:97]: why is this necessary?

**Review by johannes-huther — CHANGES_REQUESTED** [review]
Please add story

File: src/components/CCardDialog.vue (hunk 2 above)
Comment (ItsPierre) [reply src/components/CCardDialog.vue:88]: yes, the IDE gives a warning, if there is no undefined, I am not sure why. This is for every type except for the booleans

File: src/components/CCardDialog.vue (hunk 3 above)
Comment (ItsPierre) [reply src/components/CCardDialog.vue:97]: Because vuetify added a fixed margin (and idk why, but also the max-width being 100% - this margin) and doesn't reduce this on mobile. I removed it here and set our own responsive margin.

**Review by johannes-huther — CHANGES_REQUESTED** [review]
image

File: stories/CCardDialog.stories.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    loading: {
+      control: "boolean",
+    },
+  },
+  args: {
+    modelValue: false,
+  },
+  parameters: createComponentStorybookParameters({
+    componentDescription:
+      'This component is a wrapper for the vuetify dialog with a card following the material design 3 specifications. \n\n *Due to storybook restrictions all non-default stories do not work on this docs page, but they do on their respective story pages.*"',
```
Comment (johannes-huther) [inline stories/CCardDialog.stories.ts]:
```suggestion
      'This component is a wrapper for the vuetify dialog with a card following the material design 3 specifications. \n\n *Due to storybook restrictions all non-default stories do not work on this docs page, but they do on their respective story pages.*',
```

**Review by johannes-huther — CHANGES_REQUESTED** [review]

**Review by johannes-huther — APPROVED** [review]
